packages/jest-mock/src/index.ts

```diff
@@ -129,13 +130,14 @@ export interface MockInstance<T, Y extends Array<unknown>> {
   mockReturnThis(): this;
   mockReturnValue(value: T): this;
   mockReturnValueOnce(value: T): this;
-  mockResolvedValue(value: Unpromisify<T>): this;
-  mockResolvedValueOnce(value: Unpromisify<T>): this;
+  mockResolvedValue(value: Awaited<T>): this;
+  mockResolvedValueOnce(value: Awaited<T>): this;
```

Comment:
`Awaited` is introduced in TS 4.5. These are public types and Jest is aiming to support TS 4.2. So I have to revert these.

Reply:
Hah, right. Good example of #12444 😀

Reply:
somewhat sad our existent test for 4.2 didn't pick it up, but it's very much a smoke test. #12444 will be way better, especially as we expand our type tests.

Reply:
Now it got caught. I was pushing with `Awaited` just to check. All works.

Reply:
wonderful!